src/runtime/relax_vm/builtin.cc

```diff
@@ -545,6 +545,23 @@ TVM_REGISTER_GLOBAL("vm.builtin.tensor_to_shape").set_body_typed([](NDArray data
   return ShapeTuple(out_shape);
 });
 
+TVM_REGISTER_GLOBAL("vm.builtin.ensure_zero_offset").set_body_typed([](NDArray data) {
+  if (data->byte_offset == 0) {
+    return data;
+  }
+  if (DeviceAPI::SupportsPointerArithmetics(data->device.device_type)) {
```

Comment:
Can we also restrict this to `data->byte_offset % kAllocAlignment == 0`?  The alignment guarantee from TVM's allocators ensure that `NDArray->data % kAllocAlignment == 0`.  This is checked when taking ownership of a `DLManagedTensor*`, and would raise an error in `NDArray::FromDLPack` if it fails.